cmd/imagec/main.go

```diff
@@ -195,11 +200,79 @@ func main() {
 		if err := ic.ListLayers(); err != nil {
 			log.Fatalf("Listing layers for image failed due to %s\n", err)
 		}
+	case Save:
+		options := imageCOptions.options
+
+		options.Outstream = os.Stdout
+
+		ap := archiveProxy(options.Host)
+		ic := imagec.NewImageC(options, streamformatter.NewJSONStreamFormatter())
+		if err := saveImage(ap, ic, options.Reference); err != nil {
+			log.Fatalf("Saving image %s failed due to %s\n", options.Reference.String(), err)
+		}
 	default:
 		log.Errorf("The operation '%s' is not valid\n", imageCOptions.operation)
 	}
 }
 
+func saveImage(ap proxy.VicArchiveProxy, ic *imagec.ImageC, ref reference.Named) error {
```

Comment:
It doesn't appear you really need to ref here.  You mainly use it for logging and  you can get that from ic.